### PR TITLE
fix(INT-183): resolve sonar issues and move integration script to separate file

### PR DESCRIPTION
**Pull request by mihir-4116**
## PR Description

- fixed integration specific sonar code smells and reliability bugs
- move integration load script to separate file - Resolved INT-183

## Notion ticket

Ticket link

## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: src/integrations/Shynet/browser.js [hunk 1]
```diff
@@ -40,9 +42,7 @@ class Shynet {
             idempotency: this.idempotency,
             referrer: referrerName,
             location: url,
-            loadTime:
-              window.performance.timing.domContentLoadedEventEnd -
-              window.performance.timing.navigationStart,
+            loadTime: performance.now(),
```
Comment (mihir-4116) [inline src/integrations/Shynet/browser.js:45]: Tagging @aashishmalik to confirm this changes

File: src/integrations/DCMFloodlight/utils.js [hunk 2]
```diff
@@ -156,14 +157,36 @@ const buildCustomParamsUsingIntegrationsObject = (message, integrationObj) => {
...
   }
 
   return customParams;
 };
 
+/**
+ * Generate a cryptographically secure random number between 0 and 9999999999999
+ * @returns
+ */
+const getRandomNumber = () => {
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply src/integrations/DCMFloodlight/utils.js:172]: This change is not cross-browser compatible. We need to check if crypto is available on the browser. Else, fallback to the previous implementation.

**Review by anantjain45823 — COMMENTED** [review]
customerIO, Engage, Woopra, Satismeter, tiktok, Lemnisk changes LGTM

File: src/integrations/Matomo/browser.js [hunk 3]
```diff
@@ -82,65 +68,58 @@ class Matomo {
     return !!(window._paq && window._paq.push !== Array.prototype.push);
   }
 
+  pushValues(condition, values) {
```
[the remark this answers is not in the document]
Comment (mihir-4116) [reply src/integrations/Matomo/browser.js:71]: not needed as this are class methods

File: src/integrations/INTERCOM/nativeSdkLoader.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import { LOAD_ORIGIN } from '../../utils/ScriptLoader';
+
+function loadNativeSdk() {
```
[the remark this answers is not in the document]
Comment (mihir-4116) [reply src/integrations/INTERCOM/nativeSdkLoader.js]: updated

File: src/integrations/Vero/browser.js [hunk 5]
```diff
@@ -97,12 +97,12 @@ class Vero {
     logger.debug('=== In Vero track ===');
 
     const { message } = rudderElement;
-    const { event, properties } = message;
+    const { event, properties, anonymousId } = message;
```
[the remark this answers is not in the document]
Comment (mihir-4116) [reply src/integrations/Vero/browser.js]: updated

File: src/integrations/Vero/browser.js [hunk 6]
```diff
@@ -45,11 +45,11 @@ class Vero {
    * @param {Object} tags
    */
   addOrRemoveTags(message) {
-    const { integrations } = message;
-    if (integrations && integrations[NAME]) {
+    const { integrations, anonymousId } = message;
```
[the remark this answers is not in the document]
Comment (mihir-4116) [reply src/integrations/Vero/browser.js]: updated

**Comment by Gauravudia** [comment]
Mixpanel, PinterestTag, Sendinblue, Quora Pixel changes LGTM

**Comment by mihir-4116** [comment]
Microsoft Clarity, Google Ads, Sentry, June, Lotame, Chartbeat, Moengage, Sentry, Bugsnag, GA4 LGTM

**Review by koladilip — DISMISSED** [review]
Reviewed Snappixel and GA LGTM

File: src/integrations/Matomo/browser.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ItsSudip) [reply src/integrations/Matomo/browser.js:71]: got it

**Review by ItsSudip — COMMENTED** [review]
reviewed am, gtm, fullstory, intercom, VWO, matomo, launchdarkly, ga360, livechat, postaffiliatepro

File: src/integrations/Iterable/browser.js [hunk 7]
```diff
@@ -58,50 +58,39 @@ class Iterable {
...
-      } else {
-        logger.error('The JWT token was not passed, The SDK could not be initialised.');
-        return;
-      }
+    const { integrations, traits, context, userId } = message;
+    const userEmail = traits?.email || context?.traits?.email;
+
+    const jwtToken = extractJWT(integrations);
+
+    if (!jwtToken) {
```
[the remark this answers is not in the document]
Comment (yashasvibajpai) [reply src/integrations/Iterable/browser.js:80]: LGTM...

**Comment by yashasvibajpai** [comment]
Reviewed for: 
Adobe

1. Adroll
2. Hotjar
3. Iterable
4. Rockerbox
5. Vero
6. Adobe analytics
7. Keen
8. TVSquared

**Review by utsabc — APPROVED** [review]

**Comment by aashishmalik** [comment]
Reviewed for

Braze
Heap
Pendo
Lytics
Shynet

**Comment by utsabc** [comment]
Reviewed HS, Reddit Pixel, Posthog, DCM

**Review by shrouti1507 — APPROVED** [review]
reviewed Google Optimize, Linkedln Insight Tag and Facebook Pixel

File: src/integrations/Criteo/utils.js [hunk 8]
```diff
@@ -1,28 +1,29 @@
...
+  const { properties, userId, anonymousId } = message;
 
   const setEmail = {};
   const setZipcode = {};
 
   const finalRequest = [
-    { event: 'setCustomerId', id: md5(message.userId) },
-    { event: 'setRetailerVisitorId', id: md5(message.anonymousId) },
+    { event: 'setCustomerId', id: sha256(userId).toString() },
+    { event: 'setRetailerVisitorId', id: sha256(anonymousId).toString() },
```
[the remark this answers is not in the document]
Comment (shrouti1507) [reply src/integrations/Criteo/utils.js:15]: why are we changing from md5 to sha? did we confirm from the documentation?

[the remark this answers is not in the document]
Comment (shrouti1507) [reply src/integrations/Criteo/utils.js:15]: https://help.criteo.com/kb/guide/en/all-criteo-onetag-events-and-parameters-vZbzbEeY86/Steps/775825,868657,868659 <-- from here we get to know that we need md5 only. Lets revert. And also please add the doc ref link in the code . whis is https://help.criteo.com/kb/guide/en/all-criteo-onetag-events-and-parameters-vZbzbEeY86/Steps/775825

[the remark this answers is not in the document]
Comment (mihir-4116) [reply src/integrations/Criteo/utils.js:15]: got it as per doc it should be md5 only. reverting it

File: src/integrations/Criteo/utils.js [hunk 9]
```diff
@@ -87,46 +94,114 @@ const handleProductView = (message, finalPayload) => {
...
+ * @param {*} message
+ * @param {*} finalPayload
+ * @returns
+ */
+const handlingEventDuo = (message, finalPayload) => {
+  const { event, properties } = message;
+  const eventType = event.toLowerCase().trim();
+  const productInfo = getProductInfo(properties);
+
+  if (productInfo.length === 0) {
```
Comment (anantjain45823) [inline src/integrations/Criteo/utils.js:196]: If there is an error logged from `getProductInfo()` we are again logging an error which is irrelevant

**Review by utsabc — DISMISSED** [review]

**Review by ItsSudip — APPROVED** [review]

**Review by anantjain45823 — DISMISSED** [review]

**Review by ujjwal-ab — DISMISSED** [review]

**Review by yashasvibajpai — DISMISSED** [review]

**Review by sandeepdsvs — DISMISSED** [review]

**Review by MoumitaM — DISMISSED** [review]

**Review by shrouti1507 — DISMISSED** [review]

**Review by ujjwal-ab — APPROVED** [review]

**Comment by anantjain45823** [comment]
@mihir-4116 can you resolve Sonarcloud issue coming up for math.random() if it is taken care of or internally used can we please mark it resolved.

**Review by yashasvibajpai — APPROVED** [review]

**Review by MoumitaM — APPROVED** [review]

**Review by shrouti1507 — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**